Add Cloud Infrastructure Cost KPI definitions into the handbook

jasonblais opened the pull request:
From https://docs.google.com/document/d/1ON0dzi-OfPBwM5I95OLmNU7ojj0udWcmiFSaDw9XGYE/edit#

jasonblais committed:
Add Cloud Infrastructure Cost KPI definitions into the handbook

jasonblais committed:
Update SUMMARY.md

jasonblais commented on operations/research-and-development/engineering/cloud-infrastructure-cost-kpis.md:
```diff
@@ -0,0 +1,77 @@
+
+### Average Production Workspace Cost
+
+ - **Description**: The fraction which compares the amount of Production Environment costs with the number of workspaces which at a given month of the Cloud Infrastructure service.
```
Stavros & Isabel to discuss and clarify what is captured in production environment costs.

stafot replied:
@jasonblais I agree.

ialorro replied:
Discussed with Stavros. I think after the first sentence in the Description, we can add "Production Environment will be defined the environment where we host all our customers, paid and free workspaces". Just to give a general idea of what the prod environment is. Feel free to add more description if necessary to add to this definition.

jasonblais replied:
Thanks! Added

stafot commented on operations/research-and-development/engineering/cloud-infrastructure-cost-kpis.md:
```diff
@@ -0,0 +1,77 @@
+### Average Subscription Workspace Cost
+
+ - **Description**: The fraction which compares the amount of Production Environment costs with the number of workspaces that have 11 or more users which at a given month the Cloud Infrastructure service.
+ - **Formula**: ASWC = Production Environment Costs/ Paid Workspaces
```
@jasonblais Also here we need clarification. In my opinion as paid workspaces should consider the workspaces that successfully paid an invoice for the month we measure (previous month).

jasonblais replied:
Thanks! Added, let me know your thoughts

stafot commented on operations/research-and-development/engineering/cloud-infrastructure-cost-kpis.md:
```diff
@@ -0,0 +1,77 @@
+
+ - Decrease the Average Cost Per Workspace (APWC)
+ - Keep development and test environments costs (DM) below 20% of total costs
+ - Increase the WB fraction (ratio of paid workspaces relative to total workspaces)
```
Minor: Here we use WB and then WBM. Lets choose one of them as name and use the same. (My fault on the initial doc)

jasonblais replied:
Good catch, let's use "WB", updated

ialorro reviewed (commented):
This page looks great! Thank you, Jason! I have added my comment on the definition for prod environment.

jasonblais committed:
Update cloud-infrastructure-cost-kpis.md

jasonblais commented:
Also adding Justine for editor review

stafot commented on operations/research-and-development/engineering/cloud-infrastructure-cost-kpis.md:
```diff
@@ -0,0 +1,77 @@
+### Development Metric
+
+ - **Description**: The fraction which compares the amount we spend on environments related to our features testing with the total cloud infrastructure costs.
+ - **Formula**: DM = (Test Environment Costs +Dev Environment Costs)/Total AWS Cloud Costs
```
vvvvvv minor. Let's add a space after "+" symbol, to be more readable

jasonblais replied:
Good catch! I'll update

stafot reviewed (dismissed):
Great work @jasonblais.

jasonblais committed:
Add space after a "+" sign

jasonblais commented:
Unfortunately it dismissed your reviews, but I don't want to re-request it.

@justinegeffen once you've had a chance to complete editor review (not urgent), you can merge. Isabel and Stavros approved the changes previously.

justinegeffen commented on operations/research-and-development/engineering/cloud-infrastructure-cost-kpis.md:
```diff
@@ -0,0 +1,77 @@
+
+The Cloud SRE team uses certain KPIs to monitor cloud infrastructure costs from the engineering standpoint. These KPIs are used to help the business and the Cloud SRE team to set goals for cost optimization.
+
+As of July 2021, the goals for the SRE team are to
```
```suggestion
As of July 2021, the goals for the SRE team are to:
```

justinegeffen commented on operations/research-and-development/engineering/cloud-infrastructure-cost-kpis.md:
```diff
@@ -0,0 +1,77 @@
+ - Keep development and test environments costs (DM) below 20% of total costs
+ - Increase the WB fraction (ratio of paid workspaces relative to total workspaces)
+
+Below are the primary and seconday KPIs used to measure each:
```
```suggestion
Below are the primary and secondary KPIs used to measure each.
```

justinegeffen commented on operations/research-and-development/engineering/cloud-infrastructure-cost-kpis.md:
```diff
@@ -0,0 +1,77 @@
+
+### Average Production Workspace Cost
+
+ - **Description**: The fraction which compares the amount of Production Environment costs with the number of workspaces which at a given month of the Cloud Infrastructure service. Production Environment will be defined as the environment where we host all our customers, including paid and free workspaces.
```
```suggestion
 - **Description**: The fraction which compares the amount of Production Environment costs with the number of workspaces which are active in a given month of the Cloud Infrastructure service. Production Environment will be defined as the environment where we host all our customers, including paid and free workspaces.
```

justinegeffen commented on operations/research-and-development/engineering/cloud-infrastructure-cost-kpis.md:
```diff
@@ -0,0 +1,77 @@
+
+### Average Production Freemium Cost
+
+ - **Description**: Roughly is how much free workspaces cost to us. The fraction which compares the amount of Production Environment costs with the number of active free workspaces.
```
```suggestion
 - **Description**: An approximation of what free workspaces cost us. The fraction compares the amount of Production Environment costs with the number of active free workspaces.
```

justinegeffen commented on operations/research-and-development/engineering/cloud-infrastructure-cost-kpis.md:
```diff
@@ -0,0 +1,77 @@
+
+ - **Description**: Roughly is how much free workspaces cost to us. The fraction which compares the amount of Production Environment costs with the number of active free workspaces.
+ - **Formula**: APFC = Production Environment Costs/Freemium Active Workspaces
+Note: Until February 2021 we measure all workspaces as active since we introduced hibernation functionality the last 3 days of February
```
```suggestion
**Note:** Until February 2021 we measured all workspaces as active. Hibernation functionality was introduced in the last three days of February 2021.
```

justinegeffen commented on operations/research-and-development/engineering/cloud-infrastructure-cost-kpis.md:
```diff
@@ -0,0 +1,77 @@
+
+### Development Metric
+
+ - **Description**: The fraction which compares the amount we spend on environments related to our features testing with the total cloud infrastructure costs.
```
```suggestion
 - **Description**: The fraction which compares the amount we spend on environments related to our features testing with the total Cloud infrastructure costs.
```

justinegeffen commented on operations/research-and-development/engineering/cloud-infrastructure-cost-kpis.md:
```diff
@@ -0,0 +1,77 @@
+
+### Customers Platform Balance Metric
+
+ - **Description**: The fraction which compares the variable costs we have because we run the cloud Product with the costs that are directly correlated with customers usage (both paid and freemium)
```
```suggestion
 - **Description**: The fraction which compares the variable costs we have because we run the Cloud product with the costs that are directly correlated with customers' usage (both paid and freemium).
```

justinegeffen commented on operations/research-and-development/engineering/cloud-infrastructure-cost-kpis.md:
```diff
@@ -0,0 +1,77 @@
+
+### Workspaces Balance Metric
+
+ - **Description**: The fraction which compares the paid workspaces  with the total  number of workspaces which at a given month the Cloud Infrastructure service.
```
```suggestion
 - **Description**: The fraction which compares the paid workspaces with the total number of workspaces which at a given month the Cloud Infrastructure service.
```

justinegeffen replied:
This may need to be reworded as it's not clear how the comparison is related to the Cloud Infrastructure service.

jasonblais replied:
It's basically a fraction of all Cloud workspaces that are paid workspaces.

How about this simplified description?

```
The fraction of paid workspaces divided by the total number of workspaces at a given month.
```

justinegeffen commented on operations/research-and-development/engineering/cloud-infrastructure-cost-kpis.md:
```diff
@@ -0,0 +1,77 @@
+### Secondary Average Production Workspace Cost 
+
+ - **Description**: The fraction which compares the amount of Production Environment and Core Environment costs with the number of workspaces which at a given month of the Cloud Infrastructure service.
+ - **Formula**: SAPWC  = (Production Environment Costs +Core Environment Costs)/Workspaces
```
```suggestion
 - **Formula**: SAPWC  = (Production Environment Costs + Core Environment Costs)/Workspaces
```

justinegeffen commented on operations/research-and-development/engineering/cloud-infrastructure-cost-kpis.md:
```diff
@@ -0,0 +1,77 @@
+
+### Average Total Production Freemium Cost
+
+ - **Description**: Roughly is how much free workspaces cost to us. The fraction which compares the amount of Production Environment costs with the number of all free workspaces. (active and inactive)
```
```suggestion
 - **Description**: Roughly is how much free workspaces cost to us. The fraction which compares the amount of Production Environment costs with the number of all free workspaces (active and inactive).
```

justinegeffen commented on operations/research-and-development/engineering/cloud-infrastructure-cost-kpis.md:
```diff
@@ -0,0 +1,77 @@
+ - **Description**: Roughly is how much free workspaces cost to us. The fraction which compares the amount of Production Environment costs with the number of all free workspaces. (active and inactive)
+ - **Formula**: ATPFC = Production Environment Costs/Freemium Workspaces (Active and Inactive)
+
+Note: Until February 2021 we measure all workspaces as active since we introduced hibernation functionality the last 3 days of February.
```
```suggestion
**Note:** Until February 2021 we measured all workspaces as active. Hibernation functionality was introduced in the last three days of February 2021.
```

justinegeffen reviewed (commented):
Thanks for this! 

I've added some feedback and have one question about this wording: "The fraction which compares the amount of Production Environment costs with the number of active workspaces which at a given month of the Cloud Infrastructure service."

When we say "which at a given month" do we mean, within a given month? Would it be accurate to, in this case, say:
"The fraction which compares the amount of Production Environment costs with the number of active workspaces within a given month of the Cloud Infrastructure service."

Thanks. :)

jasonblais committed:
Update operations/research-and-development/engineering/cloud-infrastructure-cost-kpis.md

Co-authored-by: Justine Geffen <justinegeffen@users.noreply.github.com>

jasonblais committed:
Update operations/research-and-development/engineering/cloud-infrastructure-cost-kpis.md

Co-authored-by: Justine Geffen <justinegeffen@users.noreply.github.com>

jasonblais committed:
Update operations/research-and-development/engineering/cloud-infrastructure-cost-kpis.md

Co-authored-by: Justine Geffen <justinegeffen@users.noreply.github.com>

jasonblais committed:
Update operations/research-and-development/engineering/cloud-infrastructure-cost-kpis.md

Co-authored-by: Justine Geffen <justinegeffen@users.noreply.github.com>

jasonblais committed:
Update operations/research-and-development/engineering/cloud-infrastructure-cost-kpis.md

Co-authored-by: Justine Geffen <justinegeffen@users.noreply.github.com>

jasonblais committed:
Update operations/research-and-development/engineering/cloud-infrastructure-cost-kpis.md

Co-authored-by: Justine Geffen <justinegeffen@users.noreply.github.com>

jasonblais committed:
Update operations/research-and-development/engineering/cloud-infrastructure-cost-kpis.md

Co-authored-by: Justine Geffen <justinegeffen@users.noreply.github.com>

jasonblais committed:
Update operations/research-and-development/engineering/cloud-infrastructure-cost-kpis.md

Co-authored-by: Justine Geffen <justinegeffen@users.noreply.github.com>

jasonblais committed:
Update operations/research-and-development/engineering/cloud-infrastructure-cost-kpis.md

Co-authored-by: Justine Geffen <justinegeffen@users.noreply.github.com>

jasonblais committed:
Update operations/research-and-development/engineering/cloud-infrastructure-cost-kpis.md

Co-authored-by: Justine Geffen <justinegeffen@users.noreply.github.com>

jasonblais committed:
Update operations/research-and-development/engineering/cloud-infrastructure-cost-kpis.md

Co-authored-by: Justine Geffen <justinegeffen@users.noreply.github.com>

jasonblais committed:
Update cloud-infrastructure-cost-kpis.md

jasonblais commented:
> When we say "which at a given month" do we mean, within a given month? Would it be accurate to, in this case, say:
"The fraction which compares the amount of Production Environment costs with the number of active workspaces within a given month of the Cloud Infrastructure service."

@justinegeffen yes, that's accurate :+1:

justinegeffen committed:
Update cloud-infrastructure-cost-kpis.md

Updated the explanation for consistency and clarity to "within a given month". Fixed some formatting (spaces)
